crates/bevy_scene/src/dynamic_scene_builder.rs

```diff
@@ -131,6 +137,26 @@ impl<'w> DynamicSceneBuilder<'w> {
         drop(type_registry);
         self
     }
+
+    pub fn extract_resources(&mut self, world: &'w World) -> &mut Self {
+        let type_registry = self.type_registry.read();
+        for (component_id, _) in world.storages().resources.iter() {
+            let reflect_resource = world
+                .components()
+                .get_info(component_id)
+                .and_then(|info| info.type_id())
+                .and_then(|type_id| type_registry.get(type_id))
+                .and_then(|registration| registration.data::<ReflectResource>());
+            if let Some(reflect_resource) = reflect_resource {
+                if let Some(resource) = reflect_resource.reflect(world) {
+                    self.resources.push(resource.clone_value());
+                }
+            }
```

Comment:
Nit: Since we now have `let-else`, we can replace the `if-let` blocks with a couple more `and_then` calls:
```suggestion
            let Some(resource) = world
                .components()
                .get_info(component_id)
                .and_then(|info| info.type_id())
                .and_then(|type_id| type_registry.get(type_id))
                .and_then(|registration| registration.data::<ReflectResource>())
                .and_then(|reflect_resource| reflect_resource.reflect(world)) else {
                continue;
            };
            self.resources.push(resource.clone_value());
```
Although I suppose at that point you could also just end it in a `map`? 🤔
```suggestion
            world
                .components()
                .get_info(component_id)
                .and_then(|info| info.type_id())
                .and_then(|type_id| type_registry.get(type_id))
                .and_then(|registration| registration.data::<ReflectResource>())
                .and_then(|reflect_resource| reflect_resource.reflect(world))
                .map(|resource| self.resources.push(resource.clone_value()));
```

Eh, maybe this is all just preference. I'll leave it up to you (or anyone else) to decide how we want to format this differently (if at all).